src/services/jsTyping.ts

```diff
@@ -0,0 +1,286 @@
+﻿// Copyright (c) Microsoft. All rights reserved. Licensed under the Apache License, Version 2.0.
+// See LICENSE.txt in the project root for complete license information.
+
+/// <reference path='services.ts' />
+
+/* @internal */
+namespace ts.JsTyping {
+
+    interface TypingResolutionHost {
+        directoryExists: (path: string) => boolean;
+        fileExists: (fileName: string) => boolean;
+        readFile: (path: string, encoding?: string) => string;
+        readDirectory: (path: string, extension?: string, exclude?: string[], depth?: number) => string[];
+    };
+
+    // A map of loose file names to library names
+    // that we are confident require typings
+    let safeList: Map<string>;
+    const notFoundTypingNames: string[] = [];
+
+    function tryParseJson(jsonPath: string, host: TypingResolutionHost): any {
+        if (host.fileExists(jsonPath)) {
+            try {
+                // Strip out single-line comments
+                const contents = host.readFile(jsonPath).replace(/^\/\/(.*)$/gm, "");

```

Comment:
commandLineParser.ts is included in the services sources I believe, and it has a 'removeComments' function that is used for `tsconfig.json`. Maybe just export/reuse that, rather than have different comment handling behavior in different JSON files.